subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/ComponentVariant.java

```diff
@@ -43,6 +47,15 @@
         ImmutableList<ExcludeMetadata> getExcludes();
     }
 
+    interface DependencyConstraint {
+        String getGroup();
+
+        @Nullable
```

Comment:
Is this really nullable? If so, the we should deal with the fact elsewhere. Looks to me like we'll fail pretty quickly when constructing a `DefaultModuleComponentSelector`

Reply:
Good catch. This was an oversight -> removed.